tests/chocolatey-tests/commands/choco-upgrade.Tests.ps1

```diff
@@ -191,6 +192,73 @@ To upgrade a local, or remote file, you may use:
         }
     }
 
+    # We are marking this test as internal, as the package we need to make use
+    # of downloads a zip archive from a internal server, and the package is also
+    # only located on an internal feed.
+    Context "Upgrading non-existing package while specifying a cache location (Arg: <_>)" -ForEach '-c', '--cache', '--cachelocation', '--cache-location' -Tag Internal, LongPaths, CacheLocation {
+        BeforeAll {
+            $paths = Restore-ChocolateyInstallSnapshot
+
+            $Output = Invoke-Choco upgrade install-chocolateyzip --version 3.21.2 --confirm "--cache" "$($paths.CachePathLong)" --no-progress

```

Comment:
Same question as the install tests: shouldn't this use the passed in argument instead of `--cache`?

Reply:
See my other comment...